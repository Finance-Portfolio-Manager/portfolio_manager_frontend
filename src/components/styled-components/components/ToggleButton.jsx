/*
Carlos Galvan Jr 

ToggleButton takes in the theme object and the toggleTheme function to change theme 
*/
import React, {  useState } from "react";
import { func, string } from "prop-types";
import styled from "styled-components";

const ThemeButton = styled.button`
  background: ${({ theme }) => theme.primaryColor};
  /* border: 2px solid ${({ theme }) => theme.secondayColor}; */
  color: ${({ theme }) => theme.primaryText} !important;
  padding: 0.6rem;
  text-align: center;
  cursor: pointer;
  text-decoration: none;
  display: inline-block;
  font-size:0.8rem;
  margin: 4px 2px;
  border-radius: 3px;
`; 

const ToggleButton = ({ theme, toggleTheme }) => {
  // const [label, setLabel] = useState(window.localStorage.getItem("theme"));
  const [label, setLabel] = useState("Change Theme");
  
  const updateLabel = () => {
    toggleTheme();
    setLabel(window.localStorage.getItem("theme"));
  };

  return (
    <ThemeButton onClick={updateLabel} className="float global__blur-effect global__dropshadow">
      {label.toUpperCase()}
      {/* Change Theme */}
    </ThemeButton>
  );
};

ToggleButton.propTypes = {
  theme: string.isRequired,
  toggleTheme: func.isRequired,
};
export default ToggleButton;
