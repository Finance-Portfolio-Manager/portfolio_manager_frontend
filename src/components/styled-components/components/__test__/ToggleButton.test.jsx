import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import ToggleButton from './../ToggleButton';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';

configure({ adapter: new Adapter() });
const wrapper = mount(<ToggleButton></ToggleButton>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<ToggleButton></ToggleButton>).toJSON();
    expect(tree).toMatchSnapshot();
})

test("click toggle button", ()=>{
    //setting local storage theme
    const getItemSpy = jest.spyOn(window.localStorage, 'getItem');
    window.localStorage.setItem('theme', 'random');

    const clickFn = jest.fn();
    const component = mount(<ToggleButton theme="theme1" toggleTheme={ clickFn } label="theme"></ToggleButton>)
    const themeButton = component.find("button");
    const clickEvent = ()=>{
        themeButton.simulate('click');
    }
    expect(clickEvent).not.toThrow();
})