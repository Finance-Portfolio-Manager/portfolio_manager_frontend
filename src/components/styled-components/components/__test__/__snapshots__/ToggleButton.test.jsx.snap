// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="sc-bdnxRM ceAwFl float"
  onClick={[Function]}
>
  CHANGE THEME
</button>
`;
